language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/BallerinaLanguageServer.java

```diff
@@ -171,6 +172,8 @@ public CompletableFuture<InitializeResult> initialize(InitializeParams params) {
             experimentalClientCapabilities = new Gson().fromJson(params.getCapabilities().getExperimental().toString(),
                     HashMap.class);
         }
+        // Subscribe for document sync events
+        EventSyncPubSubHolder subscribersHolder = EventSyncPubSubHolder.getInstance(this.serverContext);
```

Comment:
What are we supposed to do with this reference. We don't have any usages of this inside this method. I suppose this is to initialize the EventSyncPubSubscriber ?

Reply:
This is to load subscribers/publishers and subscribe. We can make the `loadServices()` method public and call it at the BallerinaLanguageServer.java if needed. For now we call it inside the constructor.

Reply:
Made it a static block